src/main/java/com/microsoft/jenkins/containeragents/util/AzureContainerUtils.java

```diff
@@ -93,40 +88,28 @@ public static ListBoxModel listResourceGroupItems(String credentialsId) throws I
     }
 
     public static Azure getAzureClient(String credentialsId) {
-        AzureMsiCredentials credential = CredentialsMatchers.firstOrNull(
-                CredentialsProvider.lookupCredentials(
-                        AzureMsiCredentials.class,
-                        Jenkins.getInstance(),
-                        ACL.SYSTEM,
-                        Collections.<DomainRequirement>emptyList()),
-                CredentialsMatchers.withId(credentialsId));
-        if (credential != null) {
-            try {
-                return getAzureClient(credential.getMsiPort());
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        } else {
-            AzureCredentials.ServicePrincipal servicePrincipal = AzureCredentials.getServicePrincipal(credentialsId);
-            return getAzureClient(servicePrincipal);
-        }
+        TokenCredentialData token = getToken(credentialsId);
```

Comment:
Handle empty case? E.G. 1. Select the credential then un-select to `--- Select Azure Credentials ---`

Reply:
done